apps/desktop-wallet/src/api/addressQueries.ts

```diff
@@ -16,21 +16,70 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { AddressHash, client } from '@alephium/shared'
+import { AddressHash, client, PAGINATION_PAGE_LIMIT } from '@alephium/shared'
+import { AddressBalance, AddressTokenBalance } from '@alephium/web3/dist/src/api/api-explorer'
 import { queryOptions } from '@tanstack/react-query'
 
+import queryClient from '@/api/queryClient'
+
 interface AddressBalanceQueryProps {
   addressHash: AddressHash
   networkId: number
   latestTxHash?: string
+  previousTxHash?: string
 }
 
+const ADDRESS_BALANCE_QUERY_KEYS = ['address', 'balance']
+
 // Adding latestTxHash in queryKey ensures that we'll refetch when new txs arrive.
 // Adding networkId in queryKey ensures that switching the network we get different data.
 // TODO: Should we add explorerBackendUrl instead?
-export const addressBalanceQuery = ({ addressHash, networkId, latestTxHash }: AddressBalanceQueryProps) =>
+export const addressAlphBalanceQuery = ({
+  addressHash,
+  networkId,
+  latestTxHash,
+  previousTxHash
+}: AddressBalanceQueryProps) =>
   queryOptions({
-    queryKey: ['address', 'balance', { addressHash, latestTxHash, networkId }],
+    queryKey: [...ADDRESS_BALANCE_QUERY_KEYS, 'ALPH', { addressHash, latestTxHash, networkId }],
     queryFn: () => client.explorer.addresses.getAddressesAddressBalance(addressHash),
+    placeholderData: queryClient.getQueryData([
+      ...ADDRESS_BALANCE_QUERY_KEYS,
+      'ALPH',
+      { addressHash, latestTxHash: previousTxHash, networkId }
+    ]) as AddressBalance,
+    staleTime: Infinity
```

Comment:
Adding `placeholderData` ensures that the new query (with the new lataestTxHash in its queryKey) will be prefilled with the data of the previous one in order to avoid component re-render with `undefined` or `0` value.

If we don't do that, we'd need to remove the latest tx hash from the query keys and we'd need to find a way to invalidate these queries. I quite like the simplicity of the approach of including the latest tx hash in the query keys. And the method of adding placeholder data works well!

Reply:
I like It too!